src/material/stepper/testing/step-harness.ts

```diff
@@ -0,0 +1,134 @@
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: StepHarnessFilters = {}): HarnessPredicate<MatStepHarness> {
+    return new HarnessPredicate(MatStepHarness, options)
+        .addOption('label', options.label,
+            (harness, label) => HarnessPredicate.stringMatches(harness.getLabel(), label))
+        .addOption('selected', options.selected,
+            async (harness, selected) => (await harness.isSelected()) === selected)
+        .addOption('completed', options.completed,
+            async (harness, completed) => (await harness.isCompleted()) === completed)
+        .addOption('invalid', options.invalid,
+            async (harness, invalid) => (await harness.hasErrors()) === invalid);
+  }
+
+  /** Gets the label of the step. */
+  async getLabel(): Promise<string> {
+    return (await this.locatorFor('.mat-step-text-label')()).text();
+  }
+
+  /** Gets the `aria-label` of the step. */
+  async getAriaLabel(): Promise<string|null> {
+    return (await this.host()).getAttribute('aria-label');
+  }
+
+  /** Gets the value of the `aria-labelledby` attribute. */
+  async getAriaLabelledby(): Promise<string|null> {
+    return (await this.host()).getAttribute('aria-labelledby');
+  }
+
+  /** Whether the step is selected. */
+  async isSelected(): Promise<boolean> {
+    const host = await this.host();
+    return (await host.getAttribute('aria-selected')) === 'true';
+  }
+
+  /** Whether the step has been filled out. */
+  async isCompleted(): Promise<boolean> {
+    const state = await this._getIconState();
+    return state === 'done' || (state === 'edit' && !(await this.isSelected()));
+  }
+
+  /**
+   * Whether the step is currently showing its error state. Note that this doesn't mean that there
+   * are or aren't any invalid form controls inside the step, but that the step is showing its
+   * error-specific styling which depends on there being invalid controls, as well as the
+   * `ErrorStateMatcher` determining that an error should be shown and that the `showErrors`
+   * option was enabled through the `STEPPER_GLOBAL_OPTIONS` injection token.
+   */
+  async hasErrors(): Promise<boolean> {
+    return (await this._getIconState()) === 'error';
+  }
+
+  /** Whether the step is optional. */
+  async isOptional(): Promise<boolean> {
+    // If the node with the optional text is present, it means that the step is optional.
+    const optionalNode = await this.locatorForOptional('.mat-step-optional')();
+    return !!optionalNode;
+  }
+
+  /**
+   * Selects the given step by clicking on the label. The step may not be selected
+   * if the stepper doesn't allow it (e.g. if there are validation errors).
+   */
+  async select(): Promise<void> {
+    await (await this.host()).click();
+  }
+
+  async getChildLoader(selector: string): Promise<HarnessLoader> {
+    const contentId = await this._getContentId();
```

Comment:
why don't you instead do:
```ts
contentLoader = this.documentRootLocatorFactory.harnessLoaderFor(`#${contentId}`);

getChildLoader(...) {return this.contentLoader.getChildLoader(...); }
getAllChildLoaders(...) {return this.contentLoader.getAllChildLoaders(...); }
getHarness(...) {return this.contentLoader.getHarness(...); }
getAllHarnesses(...) {return this.contentLoader.getAllHarnesses(...); }
```

Reply:
The `contentId` is asynchronous so I have query for it when the method is executed.

Reply:
Okay nevermind, I just saw what you mean and pushed the changes. That is much cleaner 👍